src/core/algorithms/association_rules/ar_verifier/ar_verifier.cpp

```diff
@@ -0,0 +1,96 @@
+#include "ar_verifier.h"
+
+#include <chrono>
+
+#include <easylogging++.h>
+
+#include "config/ar_data/threshold/option.h"
+#include "config/ar_data/transactional_data_config.h"
+#include "config/names_and_descriptions.h"
+#include "config/option_using.h"
+#include "util/timed_invoke.h"
+
+namespace algos::ar_verifier {
+ARVerifier::ARVerifier() : Algorithm({}) {
+    RegisterOptions();
+    using namespace config::names;
+    MakeOptionsAvailable({kTable, kInputFormat});
+}
+
+void ARVerifier::RegisterOptions() {
+    DESBORDANTE_OPTION_USING;
+
+    auto check_rule_part = [this](std::list<std::string> const& string_rule_part) {
+        std::vector<std::string> const& item_names_map = transactional_data_->GetItemUniverse();
+        for (auto const& item_name : string_rule_part) {
+            auto it = std::ranges::find(item_names_map, item_name);
+            if (it == item_names_map.end()) {
+                throw config::ConfigurationError(
+                        "Item in left rule part not found in item universe: " + item_name);
+            }
+        }
+    };
+
+    auto check_rule_not_empty = [](std::list<std::string> const& string_rule_part) {
+        if (string_rule_part.empty())
+            throw config::ConfigurationError("Got an empty rule: AR verifying is meaningless.");
+    };
+
+    auto combined_rule_check = [check_rule_not_empty,
+                                check_rule_part](std::list<std::string> const& string_rule_part) {
+        check_rule_not_empty(string_rule_part);
+        check_rule_part(string_rule_part);
+    };
+
+    auto opts = config::TransactionalDataOptions(&transactional_data_params_);
+    RegisterOption(std::move(opts.input_table));
+    RegisterOption(std::move(opts.input_format));
+    RegisterOption(std::move(opts.first_column_tid));
+    RegisterOption(std::move(opts.item_column));
+    RegisterOption(std::move(opts.tid_column));
+
+    RegisterOption(config::kMinimumConfidenceOpt(&minconf_));
+    RegisterOption(config::kMinimumSupportOpt(&minconf_));
```

Comment:
Let's define options right here, because currently we use them only here (like for `kARuleLeft`, `kARuleRight`).